ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+Undistort an RGB image from a ZDF point cloud using Zivid camera intrinsics.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <iostream>
+
+void pointCloudToRGB(const Zivid::PointCloud &, cv::Mat &);
+void getCameraIntrinsics(const Zivid::Camera &, cv::Mat &, cv::Mat &);
+void displayRGB(const cv::Mat &, const std::string);
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+
+        std::cout << "Capturing a frame" << std::endl;
+        const auto frame = camera.capture();
+
+        const auto intrinsics = camera.intrinsics();
+
+        std::cout << "Setting up visualization" << std::endl;
+        Zivid::CloudVisualizer vis;
+        zivid.setDefaultComputeDevice(vis.computeDevice());
+
+        std::cout << "Displaying the point cloud" << std::endl;
+        vis.showMaximized();
+        vis.show(frame);
+        vis.resetToFit();
+
+        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+        vis.run();
+
+        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+
+        const auto pointCloud = frame.getPointCloud();
+        cv::Mat rgb((int)pointCloud.height(), (int)pointCloud.width(), CV_8UC3, cv::Scalar(0, 0, 0));
+        pointCloudToRGB(pointCloud, rgb);
+
+        std::cout << "Undistorting the RGB image" << std::endl;
+
+        cv::Size size = rgb.size();
```

Comment:
I suggest moving this variable closer to where it's used. Then the user doesn't have to keep it in mind for longer than necessary when reading the code, or wonder what it's going to be used for.

C++ Core Guidelines: [ES.21: Don't introduce a variable (or constant) before you need to use it](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es21-dont-introduce-a-variable-or-constant-before-you-need-to-use-it).

The same goes for `rgbUndistorted` and `rgbUndistortedFull` below.

Reply:
Done